src/commands/art.ts

```diff
@@ -0,0 +1,127 @@
+import { SlashCommandBuilder, SlashCommandStringOption } from "@discordjs/builders";
+import { Static } from "@sinclair/typebox";
+import { RESTPostAPIApplicationCommandsJSONBody } from "discord-api-types/v9";
+import { CommandInteraction, FileOptions } from "discord.js";
+import fetch from "node-fetch";
+import { inject, injectable } from "tsyringe";
+import { Command } from "../Command";
+import { CardSchema } from "../definitions";
+import { LocaleProvider } from "../locale";
+import { getLogger, Logger } from "../logger";
+import { Metrics } from "../metrics";
+
+@injectable()
+export class ArtCommand extends Command {
+	#logger = getLogger("command:art");
+
+	constructor(metrics: Metrics, @inject("LocaleProvider") private locales: LocaleProvider) {
+		super(metrics);
+	}
+
+	static override get meta(): RESTPostAPIApplicationCommandsJSONBody {
+		return new SlashCommandBuilder()
+			.setName("art")
+			.setDescription("Display the art for a card!")
+			.addStringOption(
+				new SlashCommandStringOption()
+					.setName("input")
+					.setDescription("The password, Konami ID, or name to search for a card.")
+					.setRequired(true)
+			)
+			.addStringOption(
+				new SlashCommandStringOption()
+					.setName("type")
+					.setDescription("Whether you're searching by password, Konami ID, or name.")
+					.setRequired(false)
+					.addChoice("Password", "password")
+					.addChoice("Konami ID", "kid")
+					.addChoice("Name", "name")
+			)
+			.toJSON();
+	}
+
+	protected override get logger(): Logger {
+		return this.#logger;
+	}
+
+	async getCard(type: "password" | "kid" | "name", input: string): Promise<Static<typeof CardSchema> | undefined> {
+		let url = `${process.env.SEARCH_API}`; // treated as string instead of string? without forbidden non-null check
+		input = encodeURIComponent(input);
+		if (type === "password") {
+			url += `/card/password/${input}`;
+		} else if (type === "kid") {
+			url += `/card/kid/${input}`;
+		} else {
+			url += `/art?name=${input}`;
+		}
+		const response = await fetch(url);
+		// 400: Bad syntax, 404: Not found
+		if (response.status === 400 || response.status === 404) {
+			return undefined;
+		}
+		// 200: OK
+		if (response.status === 200) {
+			return await response.json();
+		}
+		throw new Error((await response.json()).message);
+	}
+
+	async getArt(card: Static<typeof CardSchema>): Promise<FileOptions | undefined> {
+		const artUrl = `${process.env.IMAGE_HOST}/${card.password}.png`;
+		const response = await fetch(artUrl);
+		// 400: Bad syntax, 404: Not found
+		if (response.status === 400 || response.status === 404) {
+			return undefined;
+		}
+		// 200: OK
+		if (response.status === 200) {
+			const art = await response.buffer();
+			return { attachment: art };
+		}
+		throw new Error((await response.json()).message);
+	}
```

Comment:
So we are reuploading the artwork to Discord? I think we can do a hyperlink and let Discord do the trick. You can do just a HEAD request here to check if the art exists.

Reply:
There are two things to consider here.
The original reason I did this was "just in case" the link embed failed now and then. This is probably not reasonable to design around, though, I would concede that posting the link would generally be better.
One reason to reupload it is if we wish to obscure the URL it came from, this may not be necessary or desirable though.